data-processing/entities/abbreviations/extractor.py

```diff
@@ -0,0 +1,163 @@
+        for reserved_char in PYSBD_RESERVED_CHARACTERS:
+            if reserved_char in tex:
+                logging.warning(
+                    'Reserved character from pysbd "%s" found in tex string, this might break the sentence extractor.',
+                    reserved_char,
+                )
+
+        # Extract plaintext segments from TeX
+        plaintext_extractor = PlaintextExtractor()
+        plaintext_segments = plaintext_extractor.parse(tex_path, tex)
+
+        # Build a map from character offsets in the plaintext to TeX offsets. This will let us
+        # map from the character offsets of the sentences returned from the sentence boundary
+        # detector back to positions in the original TeX.
+        plaintext_to_tex_offset_map = {}
+        plaintext = ""
+        last_segment = None
+        for segment in plaintext_segments:
+            for i in range(len(segment.text)):
+                tex_offset = (
+                    (segment.tex_start + i)
+                    if not segment.transformed
+                    else segment.tex_start
+                )
+                plaintext_to_tex_offset_map[len(plaintext) + i] = tex_offset
+
+            # While building the map, also create a contiguous plaintext string
+            plaintext += segment.text
+            last_segment = segment
+
+        if last_segment is not None:
+            plaintext_to_tex_offset_map[len(plaintext)] = last_segment.tex_end
+
+        #this is the most basic model and had no real performance difference on our inputs
+        #other options include NER models and models with pretrained word vectors
+        nlp = spacy.load("en_core_sci_sm")
+        abbreviation_pipe = AbbreviationDetector(nlp)
+        nlp.add_pipe(abbreviation_pipe)
+
+        #these dictionaries hold abbreviated forms, their expansions, and the location of the expansions
+        #all of them use the abbreviated form as keys
+        abb_short_forms = {}
+        abb_expansions = {}
+        expanded_locations = {}
+        doc = nlp(plaintext)
+
+        #this extracts the abbreviations from the scispacy model
+        for abrv in doc._.abbreviations:
+            count = 0
+            for s in symbols:
+                count += str(abrv).count(s)
+            #count makes sure that we don't accidentally include symbols or variables
+            if count == 0:
+                abb_short_forms[str(abrv)] = [[plaintext_to_tex_offset_map[m.start()], plaintext_to_tex_offset_map[m.start() + len(str(abrv))]] for m in re.finditer(str(abrv), plaintext)]
+                abb_expansions[str(abrv)] = str(abrv._.long_form)
+                x = plaintext.find(str(abrv._.long_form))
+                if x != -1:
+                    expanded_locations[str(abrv)] = [plaintext_to_tex_offset_map[x], plaintext_to_tex_offset_map[x + len(str(abrv._.long_form))]]
+                else:
+                    expanded_locations[str(abrv)] = [0, 0]
```

Comment:
Does this seem to be working correctly? I remember there were some open questions about that before.

Reply:
It definitely works with the current setup, I can get rid of the else case for now.